pkg/kubefed2/federate/federate.go

```diff
@@ -183,32 +185,50 @@ func GetFedResources(hostConfig *rest.Config, qualifiedTypeName, qualifiedName c
 	}, nil
 }
 
-func lookupTypeDetails(hostConfig *rest.Config, typeName ctlutil.QualifiedName) (*fedv1a1.FederatedTypeConfig, error) {
-	apiResource, err := enable.LookupAPIResource(hostConfig, typeName.Name, "")
+func getTypeConfig(hostConfig *rest.Config, typeName, namespace string, outputYAML bool) (typeconfig.Interface, error) {
```

Comment:
I'm concerned that this function is doing too much.  What do you think of having one function that retrieves a type config and another that generates a type config?

In terms of the logic of choosing how to use the functions, would it make sense for the yaml path to attempt to retrieve a type config and if retrieval fails, warn that it is generating a type config?  And is there a reason to ever have the non-yaml path generate a type config?

Reply:
I think the logic and flow of the function is fine.
I did choose to lookup the federated type also (instead of retrieving the CRD) which needs the FederatedTypeName.
I can create a FederatedTypeName from the TypeName here using some strings magic, but using the typeconfig interfaces avoids duplication of code and possibility of confusion.
So we do generate typeConfig anywhich ways.

I have split the function into two and cleared out naming and comments to address your concern. The logical flow remains same.

Reply:
PS: This is further cleaned up while rebasing #663. You can choose to review that too, if you have no further comments here, as its ready for review.

Reply:
Given that yaml output may be used to create resources in the future (and not just output to the screen for demonstration purposes), I think it is potentially problematic to prefer a generated typeconfig if one is available in the API.  There is no guarantee that a given federated deployment will use the default generated typeconfig - customization is possible (i.e. the name, kind, group, and version of the federated type).

Reply:
Got your point; Updated accordingly. PTAL!